tecs/tecs.cpp

```diff
@@ -573,10 +573,34 @@ void TECS::_initialize_states(float pitch, float throttle_cruise, float baro_alt
 void TECS::_update_STE_rate_lim()
 {
 	// Calculate the specific total energy upper rate limits from the max throttle climb rate
-	_STE_rate_max = _max_climb_rate * CONSTANTS_ONE_G;
+	const float rate_max = _max_climb_rate * CONSTANTS_ONE_G;
 
 	// Calculate the specific total energy lower rate limits from the min throttle sink rate
-	_STE_rate_min = - _min_sink_rate * CONSTANTS_ONE_G;
+	const float rate_min = - _min_sink_rate * CONSTANTS_ONE_G;
+
+	// Cd_i_specific = ... assuming planar wing with elliptical lift distribution
+	_Cd_i_specific = _auw * CONSTANTS_ONE_G; //lift
+	_Cd_i_specific = _Cd_i_specific * _Cd_i_specific / (0.5f * M_PI_F * CONSTANTS_AIR_DENSITY_SEA_LEVEL_15C * _wingspan * _wingspan);
+
+	// Cd_o_specific: subtracting induced drag from total drag at a known airspeed to calculate parasitic drag
+	_Cd_o_specific = (-rate_min - _Cd_i_specific / _indicated_airspeed_trim) / (_indicated_airspeed_trim * _indicated_airspeed_trim * _indicated_airspeed_trim);
```

Comment:
Can these variables (_Cd_i_specific,  _Cd_o_specific, etc) be kept out of the class? I only see them being calculated and used in _update_STE_rate_lim().

What I'd really like to do in TECS is cleanup the unnecessary state between calls.

Reply:
Thanks for pointing that out.